contracts/secret-storage/SecretManager.sol

```diff
@@ -775,9 +779,9 @@ contract SecretManager is
                     ackTimestamp = userStorage[_secretId].ackTimestamp;
             }
 
-            if(lastAliveTimestamp <= ackTimestamp)
+            if(endTimestamp <= ackTimestamp)
                 continue;
-            uint256 storePayment = (lastAliveTimestamp - ackTimestamp) * userStorage[_secretId].sizeLimit * SECRET_STORE_FEE_RATE;
+            uint256 storePayment = (endTimestamp - ackTimestamp) * userStorage[_secretId].sizeLimit * SECRET_STORE_FEE_RATE;
```

Comment:
usdcDeposit of secret changes after alive or dead calls, is that taken into account?

storePayment signifies the expected total payment made to store for a secret (including what have been paid as part of alive check.)

Reply:
Yes.
For alive check, usdc deposit is updated only when termination condition is reached for a secret.